batcher/aligned-batcher/src/lib.rs

```diff
@@ -51,21 +51,43 @@ const ADDITIONAL_SUBMISSION_COST_PER_PROOF: u128 = 13_000;
 const CONSTANT_COST: u128 = AGGREGATOR_COST + BATCHER_SUBMISSION_BASE_COST;
 const MIN_BALANCE_PER_PROOF: u128 = ADDITIONAL_SUBMISSION_COST_PER_PROOF * 100_000_000_000; // 100 Gwei = 0.0000001 ether (high gas price)
 
+struct BatchState {
+    batch_queue: BatchQueue,
+    user_nonces: HashMap<Address, U256>,
+    user_proof_count_in_batch: HashMap<Address, u64>,
+}
+
+impl BatchState {
+    fn new() -> Self {
+        Self {
+            batch_queue: BatchQueue::new(),
+            user_nonces: HashMap::new(),
+            user_proof_count_in_batch: HashMap::new(),
+        }
+    }
+
+    fn increment_user_proof_count(&mut self, addr: &Address) -> u64 {
+        self.user_proof_count_in_batch.get(addr).unwrap_or(&0) + 1
+    }
```

Comment:
the `increment` in the name could lead to confusion since this does not update state. Let's call it something like `get_user_proof_count`, and then sum one where needed

Reply:
Solved in [913a930](https://github.com/yetanotherco/aligned_layer/pull/679/commits/913a930bff8caa6aa635af1c054ffb66fc672209)